e2e/loadbalancer_test.go

```diff
@@ -0,0 +1,269 @@
+			Type:     "LoadBalancer",
+		},
+	}
+
+	fmt.Println("Creating Service")
+	err = e2eTest.tenantClient.Create(context.TODO(), svc)
+	g.Expect(err).ShouldNot(HaveOccurred())
+
+	g.Eventually(func() string {
+		err = e2eTest.tenantClient.Get(context.TODO(), client.ObjectKeyFromObject(svc), svc)
+		if len(svc.Status.LoadBalancer.Ingress) == 0 {
+			return ""
+		}
+		return svc.Status.LoadBalancer.Ingress[0].Hostname
+	}, "5m", "5s").ShouldNot(BeEmpty())
+
+	// Cleanup
+	err = cleanUp(mirrorDeploy, svc)
+	g.Expect(err).ShouldNot(HaveOccurred())
+
+	g.Eventually(func() error {
+		return e2eTest.tenantClient.Get(context.TODO(), client.ObjectKeyFromObject(svc), svc)
+	}, "2m", "5s").ShouldNot(BeNil())
+
+}
+
+func TestLoadbalancerPublicIP(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	_, err := deployMirrorPods(e2eTest.tenantClient)
+	g.Expect(err).ShouldNot(HaveOccurred())
+
+	ip, err := getOrCreateIP(e2eTest.civo)
+	g.Expect(err).ShouldNot(HaveOccurred())
+
+	g.Eventually(func() string {
+		ip, err = e2eTest.civo.GetIP(ip.ID)
+		return ip.IP
+	}, "2m", "5s").ShouldNot(BeEmpty())
+
+	fmt.Println("Creating Service")
+	svc, err := getOrCreateSvc(e2eTest.tenantClient, ip)
+	g.Expect(err).ShouldNot(HaveOccurred())
+
+	g.Eventually(func() string {
+		err = e2eTest.tenantClient.Get(context.TODO(), client.ObjectKeyFromObject(svc), svc)
+		if len(svc.Status.LoadBalancer.Ingress) == 0 {
+			return ""
+		}
+		return svc.Status.LoadBalancer.Ingress[0].IP
+	}, "5m", "5s").Should(Equal(ip.IP))
+
+	// Unassign reserved IP
+	fmt.Println("Unassigning IP from LB")
+	svc.Annotations = nil
+	err = e2eTest.tenantClient.Update(context.TODO(), svc)
+	g.Expect(err).ShouldNot(HaveOccurred())
+
+	g.Eventually(func() string {
+		err = e2eTest.tenantClient.Get(context.TODO(), client.ObjectKeyFromObject(svc), svc)
+		if len(svc.Status.LoadBalancer.Ingress) == 0 {
+			return ""
+		}
+		return svc.Status.LoadBalancer.Ingress[0].IP
+	}, "5m", "5s").ShouldNot(Equal(ip.IP))
+
+	// To make sure an auto-assigned IP is actually assigned to the LB
+	g.Eventually(func() string {
+		err = e2eTest.tenantClient.Get(context.TODO(), client.ObjectKeyFromObject(svc), svc)
+		if len(svc.Status.LoadBalancer.Ingress) == 0 {
+			return ""
+		}
+		return svc.Status.LoadBalancer.Ingress[0].IP
+	}, "5m", "5s").ShouldNot(BeEmpty())
+
+}
+
+func cleanUp(mirrorDeploy *appsv1.Deployment, svc *corev1.Service) error {
+	err := e2eTest.tenantClient.Delete(context.TODO(), svc)
+	if err != nil {
+		return err
+	}
+
+	return e2eTest.tenantClient.Delete(context.TODO(), mirrorDeploy)
+}
+
+func getOrCreateIP(c *civogo.Client) (*civogo.IP, error) {
+	ip, err := c.FindIP("e2e-test-ip")
+	if err != nil && civogo.ZeroMatchesError.Is(err) {
+		ip, err = c.NewIP(&civogo.CreateIPRequest{
+			Name: "e2e-test-ip",
+		})
+		if err != nil {
+			return nil, err
+		}
+	} else if err != nil {
+		return nil, err
+	}
+	return ip, err
+}
+
+func getOrCreateSvc(c client.Client, ip *civogo.IP) (*corev1.Service, error) {
```

Comment:
I don't think ip should be sent as part of this method, it's not extendable as we add more annotations. 


Can we use a similar pattern as the operator e2e test where functions like this return the corev1.Service that can then be updated within the individual test (if needed) and created. 

We could then use the same sample service in the Proxy tests for example